chess/state-transition/src/stf/v1/transition.ts

```diff
@@ -348,21 +377,31 @@ async function finalizeMatch(
   lobby: IGetLobbyByIdResult,
   timer: Timer,
   newState: MatchState,
-  pool: Pool
+  pool: Pool,
+  events: Events
 ): Promise<SQLUpdate[]> {
   const matchEnvironment = extractMatchEnvironment(lobby);
 
   // Create update which sets lobby state to 'finished'
   const endMatchTuple: SQLUpdate = [endMatch, { lobby_id: lobby.lobby_id }];
 
+  // Save the final results in the final states table
+  const results = matchResults(newState, matchEnvironment, timer);
+
+  if (results[0] !== 't') {
+    events.push(
+      encodeEventForStf(precompiles.default, MatchWon, {
```

Comment:
I feel like this should probably have an interface that looks more like this

```js
encodeEventForStf({
    from: precompiles.default,
    topic: MatchWon,
    data: {
        winner: results[0] === 'w' ? matchEnvironment.user1.wallet : matchEnvironment.user2.wallet,
    }
});
```

Reply:
I updated the interface to look like this (https://github.com/PaimaStudios/paima-engine/pull/409/commits/63bb36943cabbf04732f2a5035e0411893d2b013).